internal/backend/login.go

```diff
@@ -18,10 +19,74 @@ type LoginWithTwitterInput struct {
 	AccessTokenSecret string
 }
 
+type LoginWithAppleInput struct {
+	FirstName    string
+	LastName     string
+	Email        string
+	AccessToken  string
+	RefreshToken string
+}
+
 func (t LoginWithTwitterInput) ID() string {
 	return fmt.Sprintf("%s:%s", "twitter", t.User.IDStr)
 }
 
+func (b *delphisBackend) GetOrCreateAppleUser(ctx context.Context, input LoginWithAppleInput) (*model.User, error) {
+	// The ID is the hashed email
+	hashedEmail := fmt.Sprintf("%x", sha1.Sum([]byte(input.Email)))[:36]
+
+	userProfileObj := &model.UserProfile{
+		ID:            hashedEmail,
+		DisplayName:   fmt.Sprintf("%s %s", input.FirstName, input.LastName),
+		TwitterHandle: "unknown",
```

Comment:
Since we support different login methods, would it make more sense to change this field to something like "SocialHandle"? In the case of Apple login this can be the email. Alternatively, we can create a new field like "AppleHandle", which is less scalable if we plan other login methods in the future but does not break any code.

Reply:
I used a sentinel value here. I can change it to be more unique (and a not-possible twitter handle).

Our app function is predicated on existence of twitter. I'm shoehorning in apple but I don't want to build for it -- you'll see on the app side the app isn't really functional unless the user is authed through twitter.